src/Microsoft.ML.Transforms/NAIndicatorTransform.cs

```diff
@@ -385,5 +475,272 @@ private void FillValues(int srcLength, ref VBuffer<bool> dst, List<int> indices,
             }
         }
+
+        protected override IRowMapper MakeRowMapper(ISchema schema)
+            => new Mapper(this, schema);
+
+        private sealed class Mapper : MapperBase
+        {
+            private sealed class ColInfo
+            {
+                public readonly string Name;
+                public readonly string Source;
+                public readonly ColumnType TypeSrc;
+
+                public ColInfo(string name, string source, ColumnType type)
+                {
+                    Name = name;
+                    Source = source;
+                    TypeSrc = type;
+                }
+            }
+
+            private readonly NAIndicatorTransform _parent;
+            private readonly ColInfo[] _infos;
+            private readonly ColumnType[] _types;
+            // The isNA delegates, parallel to Infos.
+            private readonly Delegate[] _isNAs;
+
+            public Mapper(NAIndicatorTransform parent, ISchema inputSchema)
+                : base(parent.Host.Register(nameof(Mapper)), parent, inputSchema)
+            {
+                _parent = parent;
+                _infos = CreateInfos(inputSchema);
+                _types = new ColumnType[_parent.ColumnPairs.Length];
+                _isNAs = new Delegate[_parent.ColumnPairs.Length];
+                for (int i = 0; i < _parent.ColumnPairs.Length; i++)
+                {
+                    var type = _infos[i].TypeSrc;
+                    if (!type.IsVector)
+                        _types[i] = BoolType.Instance;
+                    else
+                        _types[i] = new VectorType(BoolType.Instance, type.AsVector);
+                    _isNAs[i] = _parent.GetIsNADelegate(type);
+                }
+            }
+
+            private ColInfo[] CreateInfos(ISchema inputSchema)
+            {
+                Host.AssertValue(inputSchema);
+                var infos = new ColInfo[_parent.ColumnPairs.Length];
+                for (int i = 0; i < _parent.ColumnPairs.Length; i++)
+                {
+                    if (!inputSchema.TryGetColumnIndex(_parent.ColumnPairs[i].input, out int colSrc))
+                        throw Host.ExceptSchemaMismatch(nameof(inputSchema), "input", _parent.ColumnPairs[i].input);
+                    _parent.CheckInputColumn(inputSchema, i, colSrc);
+                    var type = inputSchema.GetColumnType(colSrc);
+                    infos[i] = new ColInfo(_parent.ColumnPairs[i].output, _parent.ColumnPairs[i].input, type);
+                }
+                return infos;
+            }
+
+            public override RowMapperColumnInfo[] GetOutputColumns()
+            {
+                var result = new RowMapperColumnInfo[_parent.ColumnPairs.Length];
+                for (int i = 0; i < _parent.ColumnPairs.Length; i++)
+                    result[i] = new RowMapperColumnInfo(_parent.ColumnPairs[i].output, _types[i], default);
```

Comment:
>default [](start = 98, length = 7)

more canonical is `null`. But here, you actually need to have some metadata. 

Please wait till the `Schema` PR to be merged, that'll simplify your life here.